crates/transaction-fuzzer/src/account_universe/transfer_gen.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    account_universe::{AUTransactionGen, AccountPair, AccountPairGen, AccountUniverse},
+    executor::{ExecutionResult, Executor},
+};
+use proptest::prelude::*;
+use proptest_derive::Arbitrary;
+use std::sync::Arc;
+use sui_types::{
+    error::{SuiError, UserInputError},
+    messages::{
+        ExecutionFailureStatus, ExecutionStatus, TransactionData, TransactionKind,
+        VerifiedTransaction,
+    },
+    programmable_transaction_builder::ProgrammableTransactionBuilder,
+    utils::to_sender_signed_transaction,
+};
+
+const GAS_UNIT_PRICE: u64 = 1;
+const P2P_SUCCESS_GAS_USAGE: u64 = 1000 + 1976000;
+const P2P_FAILURE_GAS_USAGE: u64 = 1000 + 988000;
+
+/// Represents a peer-to-peer transaction performed in the account universe.
+///
+/// The parameters are the minimum and maximum balances to transfer.
+#[derive(Arbitrary, Clone, Debug)]
+#[proptest(params = "(u64, u64)")]
+pub struct P2PTransferGen {
+    sender_receiver: AccountPairGen,
+    #[proptest(strategy = "params.0 ..= params.1")]
+    amount: u64,
+}
+
+impl AUTransactionGen for P2PTransferGen {
+    fn apply(
+        &self,
+        universe: &mut AccountUniverse,
+        exec: &mut Executor,
+    ) -> (VerifiedTransaction, ExecutionResult) {
+        let AccountPair {
+            account_1: sender,
+            account_2: recipient,
+            ..
+        } = self.sender_receiver.pick(universe);
+
+        let gas_object = sender.new_gas_object(exec);
+        // construct a p2p transfer of a random amount of SUI
+        let txn = {
+            let mut builder = ProgrammableTransactionBuilder::new();
+            builder.transfer_sui(recipient.initial_data.account.address, Some(self.amount));
+            builder.finish()
+        };
+        let kind = TransactionKind::ProgrammableTransaction(txn);
+        let tx_data = TransactionData::new(
+            kind,
+            sender.initial_data.account.address,
+            gas_object.compute_object_reference(),
+            P2P_SUCCESS_GAS_USAGE,
+            GAS_UNIT_PRICE,
```

Comment:
Not for this PR but later we should use the RGP instead of the dummy gas price here. Ideally there should be something similar to our cluster test's `TestContext` that gives you this type of information.